### PR TITLE
adding update_pure_parallel_wcs

### PR DESCRIPTION
**Pull request by goudfroo**
This notebook checklist has been made available to us by the [the Notebooks For All team](https://github.com/Iota-School/notebooks-for-all/blob/main/resources/event-hackathon/notebook-authoring-checklist.md).
Its purpose is to serve as a guide for both the notebook author and the technical reviewer highlighting critical aspects to consider when striving to develop an accessible and effective notebook.

### The First Cell

- [x] The title of the notebook in a first-level heading (eg. `<h1>` or `# in markdown`).
- [x] A brief description of the notebook.
- [ ] A table of contents in an [ordered list](https://www.markdownguide.org/basic-syntax/#ordered-lists) (`1., 2.,` etc. in Markdown).
- [ ] The author(s) and affiliation(s) (if relevant).
- [x] The date first published.
- [ ] The date last edited (if relevant).
- [ ] A link to the notebook's source(s) (if relevant).

### The Rest of the Cells

- [x] There is only one H1 (`#` in Markdown) used in the notebook.
- [x] The notebook uses other [heading tags](https://www.markdownguide.org/basic-syntax/#headings) in order (meaning it does not skip numbers). 

## Text

- [x] All link text is descriptive. It tells users where they will be taken if they open the link.
- [x] All acronyms are defined at least the first time they are used. 
- [x] Field-specific/specialized terms are used when needed, but not excessively.

## Code

- [x] Code sections are introduced and explained before they appear in the notebook. This can be fulfilled with a heading in a prior Markdown cell, a sentence preceding it, or a code comment in the code section.
- [x] Code has explanatory comments (if relevant). This is most important for long sections of code.
- [x] If the author has control over the syntax highlighting theme in the notebook, that theme has enough color contrast to be legible.
- [x] Code and code explanations focus on one task at a time. Unless comparison is the point of the notebook, only one method for completing the task is described at a time.

### Images

- [ ] All images (jpg, png, svgs) have an [image description](https://www.w3.org/WAI/tutorials/images/decision-tree/). This could be
    - [ ] [Alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt) (an `alt` property) 
    - [ ] [Empty alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt#decorative_images) for decorative images/images meant to be skipped (an `alt` attribute with no value)
    - [ ] Captions
    - [x] If no other options will work, the image is decribed in surrounding paragraphs.

- [ ] Any [text present in images](https://www.w3.org/WAI/WCAG21/Understanding/images-of-text.html) exists in a text form outside of the image (this can be alt text, captions, or surrounding text.)

### Visualizations

- [ ] All visualizations have an image description. Review the previous section, Images, for more information on how to add it.
- [ ] [Visualization descriptions](http://diagramcenter.org/specific-guidelines-e.html) include
    - [ ] The type of visualization (like bar chart, scatter plot, etc.)
    - [ ] Title
    - [ ] Axis labels and range
    - [ ] Key or legend
    - [ ] An explanation of the visualization's significance to the notebook (like the trend, an outlier in the data, what the author learned from it, etc.)

- [ ] All visualizations and their parts have [enough color contrast](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) ([color contrast checker](https://webaim.org/resources/contrastchecker/)) to be legible. Remember that transparent colors have lower contrast than their opaque versions.
- [ ] All visualizations [convey information with more visual cues than color coding](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html). Use text labels, patterns, or icons alongside color to achieve this.
- [ ] All visualizations have an additional way for notebook readers to access the information. Linking to the original data, including a table of the data in the same notebook, or sonifying the plot are all options.

### CODE REVIEW

**Comment by camipacifici** [comment]
Thank you for the notebook, @goudfroo !
Ping me here if you need help solving the execution issues.

**Comment by goudfroo** [comment]
OK, thanks, @camipacifici - a this point I fixed all PEP8 errors that I could. I don't understand the two remaining errors in the "notebook-style-checks", because they quote code syntax that is not in my notebook or script. I'm guessing they refer to syntax that is in scripts of packages that I'm importing. I presume fixing those is out of scope.

**Comment by goudfroo** [comment]
OK, thanks. Not a huge urgency for this one other than the fact that we have mentioned to the mission office that this was finished this week. Other than that, I'd say we're happy as long as it is released in September.

File: notebooks/cross_instrument/update_pure_parallel_wcs/update_parallel_wcs.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    crval_fix = par_aper.sci_to_sky(*crpix)
+    
+    msg = f"PureParallelUtils.update_pure_parallel_wcs: File: {file}"
+    msg += '\n' + f"PureParallelUtils.update_pure_parallel_wcs: FGS: {gs['fileName']} "
+    msg += '\n' + "PureParallelUtils.update_pure_parallel_wcs: original crval "
+    msg += f"{crval_init[0]:.7f} {crval_init[1]:.7f}"
+    msg += '\n' + "PureParallelUtils.update_pure_parallel_wcs:      new crval "
+    msg += f"{crval_fix[0]:.7f} {crval_fix[1]:.7f}"
+    msg += '\n' + "PureParallelUtils.update_pure_parallel_wcs:           dpix "
+    msg += f"{crpix[0] - crpix_init[0]:6.3f} {crpix[1] - crpix_init[1]:6.3f}"
```
Comment (haticekaratay) [inline notebooks/cross_instrument/update_pure_parallel_wcs/update_parallel_wcs.py:130]:
```suggestion
msg = (
    f"PureParallelUtils.update_pure_parallel_wcs: File: {file}\n"
    f"PureParallelUtils.update_pure_parallel_wcs: FGS: {gs['fileName']}\n"
    f"PureParallelUtils.update_pure_parallel_wcs: original crval {crval_init[0]:.7f} {crval_init[1]:.7f}\n"
    f"PureParallelUtils.update_pure_parallel_wcs:      new crval {crval_fix[0]:.7f} {crval_fix[1]:.7f}\n"
    f"PureParallelUtils.update_pure_parallel_wcs:           dpix {crpix[0] - crpix_init[0]:6.3f} {crpix[1] - crpix_init[1]:6.3f}"
)

```

File: notebooks/cross_instrument/update_pure_parallel_wcs/update_parallel_wcs.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+def log_comment(LOGFILE, comment, verbose=False, show_date=False, mode='a'):
+
+    """
+    Log a message to a file, optionally including a date tag
+    """
+    import time
+
+    if show_date:
+        msg = '# ({0})\n'.format(time.ctime())
```
Comment (haticekaratay) [inline notebooks/cross_instrument/update_pure_parallel_wcs/update_parallel_wcs.py:14]:
```suggestion
         msg = f'# ({time.ctime()})\n'
```

File: notebooks/cross_instrument/update_pure_parallel_wcs/update_parallel_wcs.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    else:
+        msg = ''
+
+    msg += '{0}\n'.format(comment)
```
Comment (haticekaratay) [inline notebooks/cross_instrument/update_pure_parallel_wcs/update_parallel_wcs.py:18]:
```suggestion
    msg += f'{comment}\n'
```

**Comment by goudfroo** [comment]
<p>I believe that would distract the reader from the purpose of the notebook, since there are quite a few imports. I plan to add a general statement like "The imports in the next cell are needed to run this notebook".</p>

---

 <a href='https://app.reviewnb.com/spacetelescope/jdat_notebooks/pull/241/discussion/#comment-2351128264'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by goudfroo** [comment]
<p>Thanks!</p>

---

 <a href='https://app.reviewnb.com/spacetelescope/jdat_notebooks/pull/241/discussion/#comment-2351128267'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by goudfroo** [comment]
<p>Thanks for the first suggestion on the possible URL errors. As to the second suggestion - I disagree, since that "imfile" needs to stay open for the next cells to work correctly. (I tried your suggestion, but it led to errors later on.)</p>

---

 <a href='https://app.reviewnb.com/spacetelescope/jdat_notebooks/pull/241/discussion/#comment-2351128272'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by goudfroo** [comment]
<p>Thanks, that was a mistake</p>

---

 <a href='https://app.reviewnb.com/spacetelescope/jdat_notebooks/pull/241/discussion/#comment-2351128275'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by goudfroo** [comment]
<p>Thanks</p>

---

 <a href='https://app.reviewnb.com/spacetelescope/jdat_notebooks/pull/241/discussion/#comment-2351128279'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by haticekaratay** [comment]
There are some style errors that need addressing. One is due to "mastquery" being imported but not used in the notebook, and the others are due to spacing. https://github.com/spacetelescope/jdat_notebooks/actions/runs/10908821325/job/30275681158?pr=241

**Comment by goudfroo** [comment]
I don't understand the two style errors that refer to spacing issues. As far as I can tell, the spacing issues mentioned are not present in any of the cells (nor in the script that is being run). Perhaps they are present in imported packages, but I cannot help that.

**Review by haticekaratay — APPROVED** [review]
Thanks for incorporating the suggestion. It looks good to me now.

**Comment by haticekaratay** [comment]
> I don't understand the two style errors that refer to spacing issues. As far as I can tell, the spacing issues mentioned are not present in any of the cells (nor in the script that is being run). Perhaps they are present in imported packages, but I cannot help that.

The error occurred due to "E303 too many blank lines" at the second and third cell. I removed them and it fixed the issue.

**Pull request merged**